lib/mutant/isolation.rb

```diff
@@ -33,30 +37,41 @@ module Fork
       #   if block terminates abnormal
       #
       # rubocop:disable MethodLength
-      def self.call(&block)
-        IO.pipe(binmode: true) do |reader, writer|
-          writer.binmode
+      def call(&block)
+        io.pipe(binmode: true) do |reader, writer|
           begin
```

Comment:
What if you pull this inner loop into a method named `parent` and gave it the same signature as `#child`? eg:

```ruby
def call(&block)
  io.pipe(binmode: true) do |reader, writer|
    parent(reader, writer, &block)
  end
rescue => exception
  raise Error, exception
end

private

def parent(reader, writer, &block)
  pid = process.fork do
    child(reader, writer, &block)
  end
  writer.close
  marshal.load(reader)
ensure
  Process.waitpid(pid) if pid
end

def child(reader, writer, &block)
  reader.close
  writer.binmode

  devnull.call do |null|
    stderr.reopen(null)
    stdout.reopen(null)
    writer.syswrite(marshal.dump(yield))
  end
  writer.close  # XXX: can this be done outside the devnull.call block?
end
```